### ISSUE TITLE
Rewriting Git history has real-world consequences

### ISSUE THREAD

**Issue by ondrejmirtes** [nette/latte#86]
There are at least two different commits representing Latte v2.3.5:

* 3d47b0af6967c99df203e98cce6201b5d84b5ea4
* 36d5f660b08f2153c2c0a99eb5f29c3b4f268698

I really don't know why that's the case. If the original 2.3.5 release contained a bug, what really should have been done was to release a new version 2.3.6 with the bugfix.

Because of this selfish change, our team cannot continue working on our application this morning. `composer install` failed during our cronjobs tonight which resulted in corrupted application on our dev server.

I propose to never forcepush to master and to never rewrite published tags.

**Comment by pavelkouril** [comment]
:+1:

**Comment by grongor** [comment]
:+1:

**Comment by hrach** [comment]
Let me rephrase:
You problem happens rarely.
Of course :+1:

**Comment by jirihelmich** [comment]
There's a nice new feature of GitHub, which is called "protected branch". Enabling that should prevent this.

**Comment by fprochazka** [comment]
:+1:

**Comment by hranicka** [comment]
:+1:

**Comment by nechutny** [comment]
:+1:

**Comment by petrjirasek** [comment]
:+1:

**Comment by dg** [comment]
I'm sorry, this was not the intention. When someone yesterday found bug in Latte, I released the new version. It seems that on Friday when I pushed on the way to the Hradec from my notebook, which I use only when I am away from home, I have older repository. 

I am forcepushing really often, but change of few days old tag is an accident.

**Comment by dg** [comment]
I am not sure if I should fix the tag to initial commit or let it go ...

**Comment by ondrejmirtes** [comment]
Just let it go and do not rewrite it again :)
